src/js/components/Meter/stories/Circle.js

```diff
@@ -1,37 +1,32 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { storiesOf } from '@storybook/react';
 
 import { Grommet, Box, Meter } from 'grommet';
 import { grommet } from 'grommet/themes';
 
-class CircleMeter extends React.Component {
-  state = { value: 20 };
+const CircleMeter = () => {
+  const [ value, setValue ] = useState(20);
+  const [ timer ] = useRef(setInterval(() => {
```

Comment:
1) haven't tested this and i might be wrong but i think useRef returns a value not an array. 
2) why are we doing a useRef with a setInterval? seems weird to me. you might want to look at this article https://overreacted.io/making-setinterval-declarative-with-react-hooks/#just-show-me-the-code

Reply:
I think you are right about useRef being wrong, it should be useState.

I initially created a useEffect with a setState, but that was one extra re-render and more convoluted. I think doing the setInterval in the initialisation of useState is fine, as it will only be called once ever, which will have the same effect as the old code.

Using timeouts and intervals though does trigger some uneasiness, especially in something like react. It may indeed be worth refactoring. A custom hook may well make sense. I resisted that as I'm not too familiar with building my own hooks and I was short on time last night. Will give it another look soon, thanks for the feedback

Reply:
@oorestisime 
I've fixed this now, ran it in storybook to verify (cool tool by the way). useState didn't work either. The problem is trickier than it first appears, as the setInterval ends up using old state values and things get confusing.

I've refactored it to use a clearTimeout and setTimeout on re-render for now. The custom hook you linked looks like the right solution if this problem crops up in the future and you need a more generic solution in multiple places.

I couldn't see an obvious location for custom hooks in the project, something to maybe think about, but beyond the scope of this pull request I feel.

Thanks for spotting the error

Reply:
hi @scfoxcode check out this example to see if it works, it loads the timer once when the component mounts https://github.com/ShimiSun/grommet-graphs/blob/f078d7aa34c64404debbf456bd5ca4bd670be04d/src/components/analytics/geo/GeographyByClient.js#L12